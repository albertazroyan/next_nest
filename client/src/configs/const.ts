import { AdminData } from '@/custom-types'
export const DefaultAdminSettingsData : AdminData = {
  book: [
    {
      author_am: 'Խոդեդան',
      author_ru: 'Ходедан',
      author_en: 'Khodedan',
      publisher_am: 'Դաշտենց Խ.',
      publisher_ru: 'Даштенц Х.',
      publisher_en: 'Dashtents Kh.',
      weight: 0.78000,
      barcode: '9789939663593',
      language: 'Հայերեն',
      newness: 'no',
      printing_cover_am: 'Կոշտ կազմ',
      printing_cover_ru: 'Жесткий состав',
      printing_cover_en: 'Hard composition',
      publication_data: 10,
      pages: 274,
      img_urls: ['uniknown.jpg']
    },
    {
      author_am: 'Խոդեդան',
      author_ru: 'Ходедан',
      author_en: 'Khodedan',
      publisher_am: 'Դաշտենց Խ.',
      publisher_ru: 'Даштенц Х.',
      publisher_en: 'Dashtents Kh.',
      weight: 0.78000,
      barcode: '9789939663593',
      language: 'Հայերեն',
      newness: 'no',
      printing_cover_am: 'Կոշտ կազմ',
      printing_cover_ru: 'Жесткий состав',
      printing_cover_en: 'Hard composition',
      publication_data: 10,
      pages: 274,
      img_urls: ['uniknown.jpg']
    }
  ]
}