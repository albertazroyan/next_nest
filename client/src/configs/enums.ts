export enum AdminTabs {
  Book = 'Book',
  Settings = 'Settings'
}