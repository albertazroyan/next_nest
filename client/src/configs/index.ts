export * from './enums'
export * from './global'