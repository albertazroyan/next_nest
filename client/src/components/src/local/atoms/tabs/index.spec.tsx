
import React from 'react'
import { render } from '@testing-library/react'
import { AdminTabs } from '../../../../../configs'

// Import the component that is being tested
import Tabs from './index'

// Define a test case
it('tabs renders correctly', () => {
  
  // Define some tabs to test with
  const contentTabs = [
    AdminTabs.Book,
    AdminTabs.Settings
  ]
  
  // Define a function to handle tab changes
  const onChange = (changedTab: string, key: number, e:  React.MouseEvent<HTMLElement> | undefined): void => {
    if (typeof changedTab === 'string') {
      console.log('changedTab:', changedTab)
      console.log('key:', key)
      console.log('e:', e)
    }
  }
  
  // Render the Tabs component with the specified tabs and onChange function
  const { container } = render(<Tabs tabs={contentTabs} onChange={onChange} />)
  
  // Expect the component to match the snapshot
  expect(container).toMatchSnapshot()
})
