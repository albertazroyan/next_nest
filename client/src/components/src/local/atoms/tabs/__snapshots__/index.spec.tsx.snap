// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabs renders 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="item active"
    >
      book
    </div>
    <div
      class="item"
    >
      settings
    </div>
  </div>
</div>
`;
